src/Serilog.Sinks.Loggly/Sinks/Loggly/FileBufferDataProvider.cs

```diff
@@ -0,0 +1,279 @@
+#endif
+            catch (Exception ex)
+            {
+                SelfLog.WriteLine("Unexpected exception while testing locked status of {0}: {1}", file, ex);
+            }
+
+            return false;
+        }
+
+        List<LogglyEvent> GetListOfEvents(string currentFile)
+        {
+            var events = new List<LogglyEvent>();
+            var count = 0;
+            var positionTracker = _currentBookmark.Position;
+
+            using (var current = _fileSystemAdapter.Open(currentFile, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
+            {
+                current.Position = positionTracker;
+
+                while (count <= _batchPostingLimit && TryReadLine(current, ref positionTracker, out string nextLine))
+                {
+                    // Count is the indicator that work was done, so advances even in the (rare) case an
+                    // oversized event is dropped.
+                    ++count;
+
+                    if (_eventBodyLimitBytes.HasValue 
+                        && nextLine != null
+                        && _encoding.GetByteCount(nextLine) > _eventBodyLimitBytes.Value)
+                    {
+                        SelfLog.WriteLine(
+                            "Event JSON representation exceeds the byte size limit of {0} and will be dropped; data: {1}",
+                            _eventBodyLimitBytes, nextLine);
+                    }
+                    if (!nextLine.StartsWith("{"))
+                    {
+                        //in some instances this can happen. TryReadLine no longer assumes a BOM if reading from the file start, 
+                        //but there may be (unobserved yet) situations where the line read is still not complete and valid 
+                        // Json. This and the try catch that follows are, therefore, attempts to preserve the 
+                        //logging functionality active, though some events may be dropped in the process.
+                        SelfLog.WriteLine(
+                            "Event JSON representation does not start with the expected '{{' character. " +
+                            "This may be related to a BOM issue in the buffer file. Event will be dropped; data: {0}",
+                            nextLine);
+                    }
+                    else
+                    {
+                        try
+                        {
+                            events.Add(DeserializeEvent(nextLine));
+                        }
+                        catch (Exception ex)
+                        {
+                            SelfLog.WriteLine(
+                                "Unable to deserialize the json event; Event will be dropped; exception: {0};  data: {1}",
+                                ex.Message, nextLine);
+                        }
+                    }
+                }
+            }
+
+            _futureBookmark = new Bookmark(positionTracker, _currentBookmark.FileName);
```

Comment:
shouldn't the positiontracker be updated to the new position?

Reply:
i now see it was updated as a ref in the TryReadLine

Reply:
positionTracker is local to the function, and is updated by `TryReadLine` (it's refed in the args list). It is currently only used to mark the position after each line read, and used to create the marker when returning the events.

`GetListOfEvents` could probably affect `_currentBatchOfEventsToProcess` directly, instead of returning (rename could be `RefreshCurrentListOfEvents`)

did you see another point of updating `positionTracker` that I may be missing?